Promote RouteAdvertisements feature gate to default feature set

jcaamano opened the pull request.

jcaamano commented:
/hold

JoelSpeed commented:
/retitle Promote RouteAdvertisements feature gate to default feature set

Retitling for consistency/searchability in the future

jcaamano commented:
/test ?

maiqueb commented:
/cc

jcaamano commented:
```
error: only 0 tests found, need at least 5 for "RouteAdvertisements" on {aws amd64 ha }
error: only 0 tests found, need at least 5 for "RouteAdvertisements" on {azure amd64 ha }
error: only 0 tests found, need at least 5 for "RouteAdvertisements" on {gcp amd64 ha }
error: only 0 tests found, need at least 5 for "RouteAdvertisements" on {vsphere amd64 ha }
```
This feature only targets metal

jcaamano commented:
```
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For cluster user defined networks When the network topology is Layer 3 UDN pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv4 [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For cluster user defined networks When the network topology is Layer 3 UDN pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv4 [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv4 [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv4 [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over a VRF-Lite configuration Pods should be able to communicate on a secondary network [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over a VRF-Lite configuration Pods should be able to communicate on a secondary network [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 3 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 3 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising the default network [apigroup:user.openshift.io][apigroup:security.openshift.io] Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising the default network [apigroup:user.openshift.io][apigroup:security.openshift.io] Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv4}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For cluster user defined networks When the network topology is Layer 3 UDN pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For cluster user defined networks When the network topology is Layer 3 UDN pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over a VRF-Lite configuration Pods should be able to communicate on a secondary network [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd[64](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_api/2225/pull-ci-openshift-api-master-verify-feature-promotion/1926984361840742400#1:build-log.txt%3A64) ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over a VRF-Lite configuration Pods should be able to communicate on a secondary network [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 2 External host should be able to query route advertised pods by the pod IP [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 2 External host should be able to query route advertised pods by the pod IP [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 2 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 2 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 3 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 3 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising the default network [apigroup:user.openshift.io][apigroup:security.openshift.io] External host should be able to query route advertised pods by the pod IP [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising the default network [apigroup:user.openshift.io][apigroup:security.openshift.io] External host should be able to query route advertised pods by the pod IP [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising the default network [apigroup:user.openshift.io][apigroup:security.openshift.io] Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only has 1 runs, need at least 14 runs for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising the default network [apigroup:user.openshift.io][apigroup:security.openshift.io] Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only passed 0%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha ipv6}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [EgressIP] Advertising EgressIP [apigroup:user.openshift.io][apigroup:security.openshift.io] For the default network Pods should have the assigned EgressIPs and EgressIPs can be created, updated and deleted [apigroup:route.openshift.io] When the network is IPv6 [Suite:openshift/conformance/parallel]" only passed 91%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha dual}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over a VRF-Lite configuration Pods should be able to communicate on a secondary network [Suite:openshift/conformance/parallel]" only passed 78%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha dual}
error: "[sig-network][OCPFeatureGate:RouteAdvertisements][Feature:RouteAdvertisements][apigroup:operator.openshift.io] when using openshift ovn-kubernetes [PodNetwork] Advertising a cluster user defined network [apigroup:user.openshift.io][apigroup:security.openshift.io] Over the default VRF When the network topology is Layer 2 Pods should communicate with external host without being SNATed [Suite:openshift/conformance/parallel]" only passed 86%, need at least 95% for "RouteAdvertisements" on {metal amd64 ha dual}
```
It seems like the tolling is picking up results from jobs that should not be running these tests. The problem is that the test itself fails for unrelated reason before it has the chance to check if it should be skipped or not. Discussed on slack
https://redhat-internal.slack.com/archives/C01CQA76KMX/p1747652922183309?thread_ts=1747390032.669289&cid=C01CQA76KMX
Seems to be a tooling pitfall.

JoelSpeed commented:
/test ci/prow/verify-feature-promotion

JoelSpeed commented:
/test verify-feature-promotion

JoelSpeed commented:
/test verify-feature-promotion

jcaamano commented:
/hold cancel

JoelSpeed commented:
/override ci/prow/verify-crd-schema

Existing issues that cannot be resolved now

/override ci/prow/verify-feature-promotion

This feature is only applicable to the metal platform and is only being tested on dual stack due to limitations in CI capacity (though there are single stack specific tests).
